src/include/ipc/trace.h

```diff
@@ -51,6 +51,22 @@ struct sof_ipc_dma_trace_posn {
 	uint32_t messages;	/* total trace messages */
 } __attribute__((packed));
 
+/** part of sof_ipc_dma_trace_filter */
+struct sof_ipc_dma_trace_filter_elem {
+	int32_t uuid_id;	/**< type id, or -1 when not important */
```

Comment:
Better to have 
```c
struct elem {
    enum type type;
    int id;
    int level;
};
```
This way adding new types means a minor ABI update

Reply:
Updated to similar key-value version.